Sources/ClientRuntime/Networking/Http/URLSession/FoundationStreamBridge.swift

```diff
@@ -182,20 +157,22 @@ class FoundationStreamBridge: NSObject, StreamDelegate {
         }
     }
 
-    private class WriteToOutputStreamResult: NSObject {
-        var data = Data()
-        var error: Error?
-    }
-
     /// Write the passed data to the output stream, using the reserved thread.
     private func writeToOutputStream(data: Data) async throws {
         try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
-            Self.queue.async {
-                let result = WriteToOutputStreamResult()
-                result.data = data
-                let selector = #selector(self.writeToOutputStreamOnThread)
-                self.perform(selector, on: Self.thread, with: result, waitUntilDone: true)
-                if let error = result.error {
+            queue.async { [self] in
+                guard !buffer.isEmpty || !data.isEmpty else { continuation.resume(); return }
+                buffer.append(data)
+                var writeCount = 0
+                buffer.withUnsafeBytes { bufferPtr in
+                    let bytePtr = bufferPtr.bindMemory(to: UInt8.self).baseAddress!
+                    writeCount = outputStream.write(bytePtr, maxLength: buffer.count)
+                }
+                if writeCount > 0 {
+                    logger.info("FoundationStreamBridge: wrote \(writeCount) bytes to request body")
+                    buffer.removeFirst(writeCount)
+                }
+                if let error = outputStream.streamError {
```

Comment:
Could move if statement on 171 to inside of the scope of withUnsafeBytes and delete line 166 that initializes writeCount. line 169 should set writeCount to 0 if it doesnt write anything. Overall this could be changed to:
```
buffer.withUnsafeBytes { bufferPtr in
        if let bytePtr = bufferPtr.bindMemory(to: UInt8.self).baseAddress {
            let writeCount = outputStream.write(bytePtr, maxLength: buffer.count)
            if writeCount > 0 {
                logger.info("FoundationStreamBridge: wrote \(writeCount) bytes to request body")
                buffer.removeFirst(writeCount)
            }
        }
    }
```

Reply:
I can't restructure the `if` into the buffer access because I can't modify `buffer` inside the block where I access its contents.